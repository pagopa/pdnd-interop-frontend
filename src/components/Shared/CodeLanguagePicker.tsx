import React from 'react'
import { Chip, Typography } from '@mui/material'
import { Box } from '@mui/system'
import { useTranslation } from 'react-i18next'

type Entry = {
  label: string
  value: string
}

type CodeLanguagePickerProps = {
  entries: Array<Entry>
  activeLang: string
  onLangUpdate: (newLang: string) => () => void
}

export const CodeLanguagePicker = ({
  entries,
  activeLang,
  onLangUpdate,
}: CodeLanguagePickerProps) => {
  const { t } = useTranslation('shared-components')

  return (
    <Box sx={{ display: 'flex' }}>
      <Typography sx={{ mr: 1 }}>{t('codeLanguagePicker.languageLabel')}: </Typography>
      <Box>
        {entries &&
          entries.map((e, i) => {
            return (
              <Chip
                sx={{ mr: 1 }}
                key={i}
                size="small"
                label={e.label}
                color={activeLang === e.value ? 'primary' : 'default'}
                onClick={onLangUpdate(e.value)}
              />
            )
          })}
      </Box>
    </Box>
  )
}
