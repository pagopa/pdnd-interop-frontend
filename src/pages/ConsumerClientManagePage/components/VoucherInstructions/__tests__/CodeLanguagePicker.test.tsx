import React from 'react'
import { CodeLanguagePicker } from '../CodeLanguagePicker'
import { render } from '@testing-library/react'
import { vi } from 'vitest'

const entriesMock = [
  { value: 'python', label: 'python' },
  { value: 'javascript', label: 'javascript' },
]

describe('CodeLanguagePicker', () => {
  it('should match snapshot', () => {
    const screen = render(
      <CodeLanguagePicker entries={entriesMock} activeLang="python" onLangUpdate={vi.fn()} />
    )
    expect(screen.baseElement).toMatchSnapshot()
  })

  it('should correctly call onLangUpdate', () => {
    const onLangUpdate = vi.fn()
    const screen = render(
      <CodeLanguagePicker entries={entriesMock} activeLang="python" onLangUpdate={onLangUpdate} />
    )
    screen.getByText('javascript').click()
    expect(onLangUpdate).toHaveBeenCalledWith('javascript')
  })
})
