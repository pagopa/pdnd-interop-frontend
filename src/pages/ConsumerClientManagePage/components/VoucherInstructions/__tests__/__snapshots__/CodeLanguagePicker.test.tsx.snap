// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CodeLanguagePicker > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="css-m69qwo-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1fbouff-MuiTypography-root"
      >
        codeLanguagePicker.languageLabel
        : 
      </p>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-hqoktk-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            python
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-aiufer-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            javascript
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
